component/src/main/java/org/wso2/extension/siddhi/execution/string/FillTemplateFunctionExtension.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.execution.string;
+
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.annotation.Parameter;
+import org.wso2.siddhi.annotation.ReturnAttribute;
+import org.wso2.siddhi.annotation.util.DataType;
+import org.wso2.siddhi.core.config.SiddhiAppContext;
+import org.wso2.siddhi.core.exception.SiddhiAppRuntimeException;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.core.executor.function.FunctionExecutor;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * replaceAll(string, regex, replacement)
+ * Replaces each substring of this string that matches the given expression with the given replacement.
+ * Accept Type(s): (STRING,STRING,STRING)
+ * Return Type(s): STRING
+ */
+
+@Extension(
+        name = "fillTemplate",
+        namespace = "str",
+        description = "Replaces the templated positions in a given template with provided strings which matches with" +
+                "the index in the template.",
+        parameters = {
+                @Parameter(name = "template",
+                        description = "The string with templated fields that needs to be filled with the given " +
+                                "strings. Templated fields should be in following the format " +
+                                "{{<INDEX>}} where INDEX is an integer. \n" +
+                                "This index is used to map the strings which are used to replace the templated fields.",
+                        type = {DataType.STRING}),
+                @Parameter(name = "replacement.strings",
+                        description = "Strings to replace the templated positions in the template.\n" +
+                                "There can be any number of strings from the 2nd argument, as comma separated values.",
+                        type = {DataType.STRING}),
+        },
+        returnAttributes =
+            @ReturnAttribute(
+                description = "String in which templated positions filled with the given strings.",
+                type = DataType.STRING),
+        examples =
+            @Example(
+                description = "" +
+                        "In this example, the templated string with index 1 {{1}} will be replaced with the 1st " +
```

Comment:
"In this example, the templated string is 'This is {{1}} for the {{2}} function', here the templated string ...